src/Generator.Bind/XML/Signatures/Functions/FunctionSignature.cs

```diff
@@ -69,7 +69,7 @@ public class FunctionSignature : INamedExtensionScopedEntity
         /// <summary>
         /// Gets a value indicating whether the function has generic type parameters.
         /// </summary>
-        public bool HasGenericTypeParameters => GenericTypeParameters.Any();
```

Comment:
Any particular reason for this? LINQ is usually cheap and more expressive. If it's for perf, I'd like to see before and after numbers.

Reply:
While this may not be performance critical code, creating an iterator and then enumerating a sequence when you can do a field comparison is just crazy.

Remember, when expanded, `Any()` is doing ALL of this underneath the hood.

```csharp
if (GenericTypeParameters == null)
{
    throw new ArgumentNullException(nameof(GenericTypeParameters));
}

var e = GenericTypeParameters.GetEnumerator();
try
{
    return e.MoveNext();
}
finally
{
    e.Dispose();
}
```
vs the vastly simpler int comparison:
```csharp
GenericTypeParameters.Count == 0
````

Reply:
I see the point, but I personally prefer the more semantically direct `Any()` call. The speed difference ought to be negligible.

Reply:
`.Count > 0` is the correct usage of the object's API. given that we're not working working with an `IReadOnlyList<T>` and not just an `IEnumerable<T>`.

Reply:
I'm a LINQ-aholic, but I think varon's got this right. We should use `Count`.

Reply:
Alright :)